lib/mayaUsd/utils/loadRulesAttribute.cpp

```diff
@@ -0,0 +1,134 @@
+//
+// Copyright 2022 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "loadRules.h"
+
+#include <maya/MCommandResult.h>
+#include <maya/MDGModifier.h>
+#include <maya/MFnDependencyNode.h>
+#include <maya/MFnTypedAttribute.h>
+#include <maya/MGlobal.h>
+#include <maya/MString.h>
+
+namespace MAYAUSD_NS_DEF {
+
+namespace {
+
+const char loadRulesAttrName[] = "usdStageLoadRules";
+
+bool hasLoadRulesAttribute(const MFnDependencyNode& depNode)
+{
+    MString nodeName = depNode.absoluteName();
+    MString cmd;
+    cmd.format("addAttr -query -exists \"^2s.^1s\"", loadRulesAttrName, nodeName);
```

Comment:
It is a shame to call MEL code in c++. How hard would it be to find the proper OpenMaya call to do the same?

Reply:
I tried initially, but the code would just crash when accessing the dynamic attribute I added. Since I had prototyped it in MEl in Maya and it worked just fine, that MEL mostly never crashes even when things are not right and that I doubt people will create many stages in a single Maya file, I don't think it is worth trying to optimize it.

Reply:
Plus, frankly, the docs in the C++ API related to dynamic attribute is... obscure or missing. Even searching the Maya code base yielded next to zero use case examples.